backend/core/src/main/java/org/sonarsource/sonarlint/core/ConfigurationService.java

```diff
@@ -100,6 +102,13 @@ public void didUpdateBinding(String configScopeId, BindingConfigurationDto updat
     }
   }
 
+  @EventListener
+  public void connectionRemoved(ConnectionConfigurationRemovedEvent event) {
+    var bindingConfigurationByConfigScope = repository.removeBindingForConnection(event.getRemovedConnectionId());
+    bindingConfigurationByConfigScope.forEach((configScope, bindingConfiguration) ->
+      applicationEventPublisher.publishEvent(new BindingConfigChangedEvent(configScope, bindingConfiguration, BindingConfiguration.noBinding())));
+  }
```

Comment:
When we automatically unbind the scope, we should still keep the value that is set in `BindingConfiguration.bindingSuggestionDisabled`. This is a user setting, and we should keep it as it is even if we force the unbind. This flag should maybe be outside of BindingConfiguration, we have [SLCORE-752](https://sonarsource.atlassian.net/browse/SLCORE-752) for that

Reply:
Addressed in [d9d5ad2](https://github.com/SonarSource/sonarlint-core/pull/982/commits/d9d5ad261521e1be145998b0fa925f68b0981f2a)